sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/connectionstring/StorageAuthenticationSettings.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation.connectionstring;
+
+import com.azure.storage.common.implementation.Constants;
+import java.util.Objects;
+
+/**
+ * Type represents settings for accessing a storage account.
+ */
+public final class StorageAuthenticationSettings {
+    private final Type type;
+    private final String sasToken;
+    private final Account account;
+
+    /**
+     * @return the settings type (None, Account Name and Key, Sas token)
+     */
+    public Type getType() {
+        return this.type;
+    }
+
+    /**
+     * @return the sas token
+     */
+    public String getSasToken() {
+        return this.sasToken;
+    }
+
+    /**
+     * @return the account instance containing account name and key
+     */
+    public Account getAccount() {
+        return this.account;
+    }
+
+    /**
+     * Creates {@link StorageAuthenticationSettings} from the given connection settings.
+     *
+     * @param settings the connection settings.
+     * @return the StorageAuthenticationSettings.
+     */
+    public static StorageAuthenticationSettings fromConnectionSettings(final ConnectionSettings settings) {
+        final String accountName = settings.getSettingValue(Constants.ConnectionStringConstants.ACCOUNT_NAME_NAME);
+        final String accountKey = settings.getSettingValue(Constants.ConnectionStringConstants.ACCOUNT_KEY_NAME);
+        final String sasSignature =
+                settings.getSettingValue(Constants.ConnectionStringConstants.SHARED_ACCESS_SIGNATURE_NAME);
+
+        if (accountName != null && accountKey != null && sasSignature == null) {
```

Comment:
If account name and key not null, then get into the block. No need to check sas, right?

Reply:
Actually as per validation rules - {account name:key} and {Sas} are mutually exclusive, we do that validation already on connection string, here it is just reinforcing for readability.